src/codegen.cpp

```diff
@@ -1775,6 +1783,7 @@ static void cg_bdw(jl_binding_t *b, jl_codectx_t *ctx)
 // try to statically evaluate, NULL if not possible
 static jl_value_t *static_eval(jl_value_t *ex, jl_codectx_t *ctx, int sparams=true, int allow_alloc=true)
 {
+    if (!JL_FEAT_TEST(ctx, static_alloc)) allow_alloc = 0;
```

Comment:
`allow_alloc` sounds ambiguous, maybe not in the context of `static_eval` but on its own as a codegen param. Also, to contrast against `dynamic_alloc`...